File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int compute_grade(int letters, int words, int sentences);

int main(void)
{
    // get input
    string word1 = get_string("Text: ");

    // only word count should start from 1
    // word count = white spaces + 1
    int wordCount = 1;
    int letterCount = 0;
    int sentenceCount = 0;
    // loop over text
    for (int i = 0; i < strlen(word1); i++) {
        // check word ending punctuations
        if (word1[i] == '!' || word1[i] == '?' || word1[i] == '.')
            sentenceCount++;
        // check word counts with white spaces
        if (word1[i] == ' ')
            wordCount++;

        //check if it is a alphabetical character
        if (isalpha(word1[i]))
            letterCount++;
    }

    // compute result
    int result = compute_grade(letterCount, wordCount, sentenceCount);

    // print final result
    if (result < 1)
        printf("Before Grade 1\n");
    else if (result > 16)
        printf("Grade 16+\n");
    else
        printf("Grade %d\n", result);


    return 0;
}

// calculate result with Coleman-Liau formula
int compute_grade(int letters, int words, int sentences)
{
    double l = 100 * (double) letters / (double) words;
    double s = 100 * (double) sentences / (double) words;
    double index = 0.0588 * (double) l - 0.296 * (double) s - 15.8;

    return round(index);
}

File: sentimental-readability/readability.py
import re


def main():
    # get input
    text = input("Text:")

    # only word count should start from 1
    # word count = white spaces + 1
    wordCount = 1
    letterCount = 0
    sentenceCount = 0

    # iterate through all characters on input
    for letter in text:
        if letter == '!' or letter == '?' or letter == '.':
            sentenceCount += 1

        if letter == " ":
            wordCount += 1

        isSymbol = re.sub(r'[^\w]', '', letter)
        if (isSymbol):
            letterCount += 1

    result = compute_grade(letterCount, wordCount, sentenceCount)

    # print final result
    if (result < 1):
        print("Before Grade 1\n")
    elif (result > 16):
        print("Grade 16+\n")
    else:
        print("Grade \n", result)

# apply Coleman-Liau formula


def compute_grade(letters, words, sentences):
    l = 100 * float(letters) / float(words)
    s = 100 * float(sentences) / float(words)
    index = 0.0588 * l - 0.296 * s - 15.8

    return round(index)


main()
